src/transformers/modeling_flax_pytorch_utils.py

```diff
@@ -319,11 +344,15 @@ def load_flax_checkpoint_in_pytorch_model(model, flax_checkpoint_path):
     flax_cls = getattr(transformers, "Flax" + model.__class__.__name__)
 
     # load flax weight dict
-    with open(flax_checkpoint_path, "rb") as state_f:
-        try:
-            flax_state_dict = from_bytes(flax_cls, state_f.read())
-        except UnpicklingError:
-            raise EnvironmentError(f"Unable to convert {flax_checkpoint_path} to Flax deserializable object. ")
+    if flax_checkpoint_path.endswith(".safetensors"):
+        flax_state_dict = safe_load_file(flax_checkpoint_path)
+        flax_state_dict = unflatten_dict({tuple(k.split(".")): v for k, v in flax_state_dict.items()})
```

Comment:
Equivalent to:
```suggestion
        flax_state_dict = unflatten_dict(flax_state_dict, sep=".")
```

Reply:
wooo this is a very cool method. Thanks, that's a nice tip!